### PR TITLE
69 add repositories tests

### LINKED ISSUE #69: Add repositories tests

### PR DESCRIPTION
**Pull request by Win-ther**
### Related Issue(s)
Closes #69 

### Proposed Changes
- Add tests for UserRepository and MessageRepositort
- Removed Initializer for DB, made tests not funtion as they should.
- Refactored a method in UserRepository to fit with other methods.

### Description
Testing for repositories.

### CODE REVIEW

File: src/test/java/se/iths/springbootgroupproject/repositories/MessageRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        var message = new Message();
+        message.setTitle("TEST-TITLE");
+        entityManager.persist(message);
+        String newTitle = "UPDATED TEST-TITLE";
+
+        message.setTitle(newTitle);
+        messageRepository.save(message);
+
+        assertThat(entityManager.find(Message.class, message.getId()).getTitle()).isEqualTo(newTitle);
+    }
```
Comment (Frlundsten) [inline src/test/java/se/iths/springbootgroupproject/repositories/MessageRepositoryTest.java]: I do not think this updates a user 😉

**Review by Frlundsten — CHANGES_REQUESTED** [review]
Great job! It is nice to finally have some tests 😄 I found a typo that would be nice to fix for clarity.

File: src/test/java/se/iths/springbootgroupproject/repositories/MessageRepositoryTest.java (hunk 1 above)
Comment (Win-ther) [reply src/test/java/se/iths/springbootgroupproject/repositories/MessageRepositoryTest.java]: Fixed^^

**Comment by marcusju23** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=Win-ther_SpringbootGroupProject&pullRequest=77) **Quality Gate failed**
> 
> Failed conditions  [0.0% Coverage on New Code](https://sonarcloud.io/component_measures?id=Win-ther_SpringbootGroupProject&pullRequest=77&metric=new_coverage&view=list) (required ≥ 80%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=Win-ther_SpringbootGroupProject&pullRequest=77)

@Win-ther 0.0% coverage due to integrated tests? Perhaps an adjustment in the pipeline configuration and dependencies could resolve this? 🤔 But that's for another issue if needed 😄

**Review by marcusju23 — APPROVED** [review]
Good stuff! 🙌 Just one thing.

> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=Win-ther_SpringbootGroupProject&pullRequest=77) **Quality Gate failed**
> 
> Failed conditions  [0.0% Coverage on New Code](https://sonarcloud.io/component_measures?id=Win-ther_SpringbootGroupProject&pullRequest=77&metric=new_coverage&view=list) (required ≥ 80%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=Win-ther_SpringbootGroupProject&pullRequest=77)

@Win-ther 0.0% coverage due to integrated tests? Perhaps an adjustment in the pipeline configuration and dependencies could resolve this? 🤔 But that's for another [issue](https://github.com/Win-ther/SpringbootGroupProject/issues/80) if needed 😄

**Comment by Win-ther** [comment]
> Good stuff! 🙌 Just one thing.
> 
> > ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=Win-ther_SpringbootGroupProject&pullRequest=77) **Quality Gate failed**
> > Failed conditions  [0.0% Coverage on New Code](https://sonarcloud.io/component_measures?id=Win-ther_SpringbootGroupProject&pullRequest=77&metric=new_coverage&view=list) (required ≥ 80%)
> > [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=Win-ther_SpringbootGroupProject&pullRequest=77)
> 
> @Win-ther 0.0% coverage due to integrated tests? Perhaps an adjustment in the pipeline configuration and dependencies could resolve this? 🤔 But that's for another [issue](https://github.com/Win-ther/SpringbootGroupProject/issues/80) if needed 😄

The quality-gate failing with 0.0% coverage on new code is because I added code to the service class. If you click on the failed analysis you can see that it only mentions UserService.
image
The analsysis seems to not care about the Repositories, even in PR #3 the analysis does not react to the repositories.

It doesn't react to the Integration tests, because they do not add coverage, and the coverage is 100% by default for the repositories since they don't have real methods. They are interfaces.
Because of that I see no reason to add an Issue(#80) for it since there is nothing to fix. Or am I missing something? 🙂

**Comment by marcusju23** [comment]
> The quality-gate failing with 0.0% coverage on new code is because I added code to the service class. If you click on the failed analysis you can see that it only mentions UserService.
>The analsysis seems to not care about the Repositories, even in PR #3 the analysis does not react to the repositories.

Ah okay, I somehow missed that. I'll close #80

**Review by Frlundsten — APPROVED** [review]
Thank you for your service, 10/10

**Review by altmannk — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**